repos/system_upgrade/el7toel8/actors/addupgradebootentry/actor.py

```diff
@@ -21,4 +23,12 @@ def process(self):
         if grub_config_error_detected:
             fix_grub_config_error('/etc/default/grub')
 
-        add_boot_entry()
+        config = None
+        ff = next(self.consume(FirmwareFacts), None)
+
+        # related to issue with hybrid BIOS and UEFI images
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1667028
+        if ff and ff.firmware == 'bios' and os.path.ismount('/boot/efi'):
```

Comment:
it's nitpick, nothing searious, but `FirmwareFacts` should be produced always, unless the code is buggy. Without the message, how you could tell what is the config? (ignoring manual check of `/sys/firmware/efi`). I would rather calculate with the fact that firmware has to be consumed always.

Reply:
IIUC you would prefer just:
`if ff.firmware == 'bios'` and then failing with None type has no ... ?